controller/eks-cluster-config-handler.go

```diff
@@ -1151,6 +1225,13 @@ func (h *Handler) importCluster(config *eksv1.EKSClusterConfig, eksService *eks.
 		}
 	}
 
+	launchTemplatesOutput, err := ec2Service.DescribeLaunchTemplates(&ec2.DescribeLaunchTemplatesInput{
+		LaunchTemplateNames: []*string{aws.String(fmt.Sprintf(launchTemplateNameFormat, config.Spec.DisplayName))},
```

Comment:
Is this checking to see if it is using a launch template that is managed by Rancher? Could this situation occur? How would the cluster become detached from rancher without being deleted? Or is this for a different use case?

Reply:
If the user imports a cluster, creates a new node group (which would create the launch template), removes the cluster from Rancher, and then imports it again: then the launch template would still be there and Rancher wouldn't be able to create a new launch template with the prescribed name.

Therefore, if a launch template with the prescribed name exist on import, the operator will get its ID so it can be managed by Rancher.

Reply:
The problem wasn't whether it did it in a single call, it just looked like it would not upgrade thee kubernetes version if there was a launch template.

Reply:
Okay that makes sense. I suppose we can't clean it up and we still don't want them managing it as their own since we need it for the other nodegroups. This does remind me, is there a reason why we don't make a template per nodegroup?

Reply:
I believe the main reason was limits: 5,000 launch templates, but 10,000 versions per template. Another reason: it is more straightforward to tell if the launch template is rancher-managed because it will have the correct template id.